apps/omg_eth/test/support/wait_for.ex

```diff
@@ -52,18 +51,25 @@ defmodule Support.WaitFor do
       end
     end
 
+    WaitFor.ok(f, timeout)
+  end
+
+  # Repeats f until f returns {:ok, ...}, :ok OR exception is raised (see :erlang.exit, :erlang.error) OR timeout
+  # after `timeout` milliseconds specified
+  #
+  # Simple throws and :badmatch are treated as signals to repeat
+  def ok(f, timeout \\ 5_000) do
```

Comment:
Aha - I finally got it :exclamation: 
This is super clever - much better, comparing to the prev implementation

Reply:
:) I didn't really change anything, just now the timeout is included, doesn't need to be provided by caller. All cases did the same thing always :).

Reply:
but previously we had some expectations of the return `:repeat` :arrow_forward: ward: it's cleaner now